### PR TITLE
feat: Calculate `defaultProtocol` for a user - WPB-14667

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-14667" title="WPB-14667" target="_blank">BugWPB-14667</a>  [iOS] Default protocol if MLS is disabled
  </td></table>
  <br />
 

### Issue

We use `defaultProtocol` to create a conversation. If a team has a weird setup, it takes the wrong `defaultProtocol`.
Example:
- mls feature is **disabled**
- but the default protocol from the feature config is MLS 

In this example, the self client is not an MLS client, but is trying to create an MLS conversation and fails.

### Solution

If mls feature is disabled then the default protocol should be `proteus`. Otherwise, it should be taken from the feature config.

### Testing

Setup a team as:

```
"mls": {
        "config": {
            "allowedCipherSuites": [
                2
            ],
            "defaultCipherSuite": 2,
            "defaultProtocol": "mls",
            "protocolToggleUsers": [],
            "supportedProtocols": [
                "mls"
            ]
        },
        "lockStatus": "unlocked",
        "status": "disabled",
        "ttl": "unlimited"
    },

```
### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift [hunk 1]
```diff
@@ -500,15 +500,20 @@ extension ConversationCreationController {
...
+        switch mlsFeature.config.defaultProtocol {
         case .proteus:
-            .proteus
+            return .proteus
         case .mls:
-            .mls
+            return .mls
         case .mixed:
-            .mixed
+            return .mixed
```
Comment (johnxnguyen) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift]: `.mixed` doesn't make sense as a default protocol. We had two separate message protocol types (one with mixed, one without) for the reason that one represents what the current protocol of a conversation is (which can be mixed) and the other represents what different protocols there are (without mixed).

The value should be either proteus or mls.

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift [hunk 2]
```diff
@@ -500,15 +500,20 @@ extension ConversationCreationController {
...
-extension Feature.MLS.Config.MessageProtocol {
-    var toMessageProtocol: MessageProtocol {
-        switch self {
+extension UserSession {
+
+    var defaultProtocol: MessageProtocol {
+        guard mlsFeature.isEnabled else {
+            return .proteus
+        }
+        switch mlsFeature.config.defaultProtocol {
```
Comment (jullianm) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift:505]: I think here you can remove the switch and directly return `mlsFeature.config.defaultProtocol.toMessageProtocol`

**Review by jullianm — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift (hunk 2 above)
Comment (KaterinaWire) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift:505]: Unfortunately, I can't. 
mlsFeature.config.defaultProtocol is `Feature.MLS.Config.MessageProtocol `type. But I would like to have a result as `MessageProtocol` enum type.

Comment (jullianm) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift:505]: isn't that the case here ?

```
extension Feature.MLS.Config.MessageProtocol {
    var toMessageProtocol: MessageProtocol {
        switch self {
        case .proteus:
            .proteus
        case .mls:
            .mls
        case .mixed:
            .mixed
        }
    }
}
```

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift [hunk 3]
```diff
@@ -500,15 +500,20 @@ extension ConversationCreationController {
     }
 }
 
-extension Feature.MLS.Config.MessageProtocol {
-    var toMessageProtocol: MessageProtocol {
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift:504]: @jullianm I removed `toMessageProtocol`, and have only one `defaultProtocol` helper property

Comment (jullianm) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift:504]: Ok, sorry I thought this was legacy code

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift (hunk 1 above)
Comment (KaterinaWire) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/ConversationCreationController.swift]: Changed it to Proteus

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**